pkg/apis/batch/validation/validation.go

```diff
@@ -400,6 +405,35 @@ func validateJobStatus(status *batch.JobStatus, fldPath *field.Path) field.Error
 	if status.Terminating != nil {
 		allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(*status.Terminating), fldPath.Child("terminating"))...)
 	}
+
+	// A strengthened validation of the Job status is needed since the
+	// Job managed-by label let's the Job object be controlled by external
+	// controllers. We want to make sure the status set by the external
+	// controllers meet the expectations of the clients of the Job API.
+	// In particular, we don't allow to set both Complete and Failed conditions.
```

Comment:
```suggestion
	// For example, we don't allow to set both Complete and Failed conditions.
```

Otherwise you would have to list all the validations added.

Reply:
done